services/core-web/src/components/common/DocumentTable.tsx

```diff
@@ -49,6 +58,11 @@ interface DocumentTableProps {
   additionalColumnProps: { key: string; colProps: any }[];
   fileOperationPermissionMap: { operation: FileOperations; permission: string | boolean }[];
   userRoles: string[];
+  project?: IProject;
```

Comment:
Would suggest a few things for this file:
1. Refactor to not need the project passed in. Looks like we need the mine guid from it (stored in every document) and the text to show in the download, which will vary depending on where we're using the table (I think we had a prop called document parent that we used for something similar?). Could create a function in the document model like getParentEntityName too, which will be easy to extend for different file types and reusable everywhere.
2. Get the permission from the document itself instead of doing a role check in here. If we want to download the file, check if the file has permission to download, etc. Basically we want the document table to be as agnostic as possible when it comes to what type of files it's displaying.
3. This file is starting to get excessively long- can anything be extracted to its own file? Something like the progress bar or modal could be good candidates.

Reply:
Great suggestions.

1. Project is no longer passed in as a prop now, and instead project title is setup inside the mine document type to be retrieved for use.

2. Added a multi select permission to the mine document type, and is checking each document in the table to make sure they all have multi select permission. No longer retrieving role for check here now.

3. The progress bar and modals are now extracted out.